src/lib.rs

```diff
@@ -290,17 +298,40 @@ impl Build {
             .unwrap_or_else(|| env::var("TARGET").expect("TARGET must be set"))
     }
 
+    /// Returns version string if nasm is too old,
+    /// or error message string if it's unusable.
+    fn is_nasm_new_enough(&self, nasm_path: &Path) -> Result<(), String> {
+        let version = get_output(Command::new(nasm_path).arg("-v"))?;
+        let (major, minor, micro) = self.min_version;
+        let ver: Vec<usize> = version
+            .split(" ")
+            .skip(2)
+            .next()
+            .map(|ver| {
+                ver.split(".")
+                    .map(|v| v.parse().expect("Invalid version component"))
+                    .collect()
+            })
+            .expect("Invalid version");
+        if ver.len() < 3 {
+            panic!("Not enough version components");
+        } else {
+            if major > ver[0] ||
+            (major == ver[0] && minor > ver[1]) ||
+             (major == ver[0] && minor == ver[1] && micro > ver[2]) {
+                Err(version)?
+            } else {
+                Ok(())
+            }
+        }
```

Comment:
Since the previous if statement `panic!`s, this code doesn't need an explicit else branch,